prime-router/src/main/kotlin/Report.kt

```diff
@@ -64,6 +65,11 @@ class Report {
      */
     val name: String get() = formFileName(id, schema.baseName, destination?.format, createdDateTime)
 
+    /**
+     * A pointer to where the Report is stored.
+     */
+    var bodyURL: String = ""
+
```

Comment:
The ActionHistory uses a more robust style of lineage tracking - rather than "insert as things happen", it accumulates a set of changes associated with an action, then saves them all at once.   

As a minor side effect of that significant stylistic change, it was convenient to have a report know its own blob storage url.